### PR TITLE
src: v2.1.9

### PR DESCRIPTION
**Pull request by vladfrangu**

### CODE REVIEW

File: src/main/kotlin/stickyWallet/StickyWallet.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            server.pluginManager.disablePlugin(this)
+            return
+        }
+
+        server.pluginManager.registerEvents(JoinLeaveListeners(), this)
+        server.pluginManager.registerEvents(CraftEvent(), this)
+
+        val balCommand = getCommand("balance")!!
+        balCommand.setExecutor(BalanceCommand())
+        balCommand.usage = "[player]"
```
Comment (kaylendog) [inline src/main/kotlin/stickyWallet/StickyWallet.kt:66]: Is it easier to set command usage in the `spigot` block in the build script? Or was that glitchy idk

File: src/main/kotlin/stickyWallet/accounts/ConsoleAccount.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+class ConsoleAccount : Account(
+    UUID.fromString("ddd22521-4c13-4239-93a5-a976bcfc614f"),
+    CheckSettings.consoleName
+) {
+    override fun hasEnough(currency: Currency, amount: BigDecimal) = true
+    override fun hasEnough(amount: BigDecimal) = true
+
+    override fun withdraw(currency: Currency, amount: BigDecimal) = true
+}
```
Comment (kaylendog) [inline src/main/kotlin/stickyWallet/accounts/ConsoleAccount.kt]: smh newline plsss <3

File: src/main/kotlin/stickyWallet/check/CheckManager.kt [hunk 3]
```diff
@@ -1,113 +1,101 @@
...
+            checkIssuer
+        )
+    }
+
+    data class CheckData(
+        val value: BigDecimal,
+        val currency: String,
+        val issuer: String
+    )
+}
```
Comment (kaylendog) [inline src/main/kotlin/stickyWallet/check/CheckManager.kt]: <3

File: src/main/kotlin/stickyWallet/currencies/Currency.kt [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                this.singular.replace("_", " ")
+            } else {
+                this.plural.replace("_", " ")
+            })
+
+        return final.toString()
+    }
+
+    fun nameEquals(input: String) = singular.equals(input, true) || plural.equals(input, true)
+}
```
Comment (kaylendog) [inline src/main/kotlin/stickyWallet/currencies/Currency.kt]: am i being too picky? i gotta pick on something hhhhh

File: src/main/kotlin/stickyWallet/StickyWallet.kt [hunk 5]
```diff
@@ -0,0 +1,130 @@
+package stickyWallet
```
Comment (kaylendog) [inline src/main/kotlin/stickyWallet/StickyWallet.kt:1]: Not a conventional package name but who cares if it works ig

File: src/main/kotlin/stickyWallet/currencies/CurrencyStore.kt [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+        return true
+    }
+
+    fun addCachedCurrency(currency: Currency) = if (currencies.contains(currency)) {
+        false
+    } else {
+        currencies.add(currency)
+    }
+}
```
Comment (kaylendog) [inline src/main/kotlin/stickyWallet/currencies/CurrencyStore.kt]: bark bark ur gay

File: src/main/kotlin/stickyWallet/interfaces/DataHandler.kt [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    abstract fun loadCurrencies(): Unit
+    abstract fun saveCurrency(currency: Currency): Unit
+    abstract fun deleteCurrency(currency: Currency): Unit
+    abstract fun updateCachedCurrency(currency: Currency): Unit
+
+    abstract fun loadAccount(playerName: String): Account?
+    abstract fun loadAccount(uuid: UUID): Account?
+    abstract fun saveAccount(account: Account): Unit
+    abstract fun deleteAccount(account: Account): Unit
+}
```
Comment (kaylendog) [inline src/main/kotlin/stickyWallet/interfaces/DataHandler.kt]: I'm surprised ktlint didn't pick up on these, did for me hhh

File: src/main/kotlin/stickyWallet/listeners/CraftEvent.kt [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+                )
+            }
+        }
+    }
+
+    private fun checkItemStackIsCheck(input: ItemStack): Boolean {
+        CheckManager.validateCheck(input) ?: return false
+        return true
+    }
+}
```
Comment (kaylendog) [inline src/main/kotlin/stickyWallet/listeners/CraftEvent.kt]: vladdy is soft
vladdy is soft
vladdy is soft

File: src/main/kotlin/stickyWallet/listeners/ItemUse.kt [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    fun onItemUse(event: PlayerInteractEvent) {
+        val player = event.player
+
+        val item = event.item ?: return
+
+        if (item.type == CheckSettings.material) {
+            player.sendActionBar("You have a check-like thing owo")
+        }
+    }
+}
```
Comment (kaylendog) [inline src/main/kotlin/stickyWallet/listeners/ItemUse.kt]: Ok - I'm not gonna annoy you anymore, you get the drill uwu

File: src/main/kotlin/stickyWallet/sql/tables/AccountsTable.kt [hunk 10]
```diff
@@ -0,0 +1,13 @@
+package stickyWallet.sql.tables
+
+import org.jetbrains.exposed.sql.Table
+import stickyWallet.configs.PluginConfiguration
+
+object AccountsTable : Table(
+    PluginConfiguration.StorageSettings.storageTablePrefix + "_accounts"
+) {
+    val playerName = varchar("player_name", 255)
+    val playerUUID = varchar("player_uuid", 255)
```
Comment (kaylendog) [inline src/main/kotlin/stickyWallet/sql/tables/AccountsTable.kt:10]: Could be more specific with field lengths here, if i'm not wrong usernames are max 16 chars and uuids are max 36

File: src/main/kotlin/stickyWallet/sql/tables/BalancesTable.kt [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+import org.jetbrains.exposed.sql.Table
+import stickyWallet.configs.PluginConfiguration
+import stickyWallet.sql.PostgresDecimal
+import java.math.BigDecimal
+
+object BalancesTable : Table (
+  PluginConfiguration.StorageSettings.storageTablePrefix + "_balances"
+) {
+    val accountID = varchar("account_id", 255)
+    val currencyID = varchar("currency_id", 255)
```
Comment (kaylendog) [inline src/main/kotlin/stickyWallet/sql/tables/BalancesTable.kt:12]: See above

File: src/main/kotlin/stickyWallet/sql/tables/CurrenciesTable.kt [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+import org.jetbrains.exposed.sql.Table
+import stickyWallet.configs.PluginConfiguration
+import stickyWallet.sql.PostgresDecimal
+import java.math.BigDecimal
+
+object CurrenciesTable : Table(
+    PluginConfiguration.StorageSettings.storageTablePrefix + "_currencies"
+) {
+    val id = integer("id").autoIncrement()
+    val uuid = varchar("uuid", 255)
```
Comment (kaylendog) [inline src/main/kotlin/stickyWallet/sql/tables/CurrenciesTable.kt:13]: See above

File: stickyWallet/files/Configuration.kt [hunk 13]
```diff
@@ -1,151 +1,120 @@
...
-        plugin.reloadConfig()
-    }
-
-    companion object {
-        private const val path = "messages."
-    }
-
-}
\ No newline at end of file
+package stickyWallet.files
```
Comment (kaylendog) [inline stickyWallet/files/Configuration.kt:1]: smh what is this doing here

File: stickyWallet/files/L.kt [hunk 14]
```diff
@@ -1,205 +1,170 @@
...
-    val exchangeRateSet
-        get() = "$prefix${get("$path.exchange_rate_set")}"
-
-    val exchangeNoPermsCustom
-        get() = "$prefix${get("$path.exchange_command.no_perms.custom")}"
-
-    val exchangeNoPermsPreset
-        get() = "$prefix${get("$path.exchange_command.no_perms.preset")}"
-}
+package stickyWallet.files
```
Comment (kaylendog) [inline stickyWallet/files/L.kt:1]: this too

File: stickyWallet/files/ass.txt [hunk 15]
```diff
@@ -0,0 +1,50 @@
+messages:
```
Comment (kaylendog) [inline stickyWallet/files/ass.txt:1]: yes vlad i know you're gay, you don't need to tell me <3

**Review by kaylendog — COMMENTED** [review]
:heart:

File: stickyWallet/files/Configuration.kt (hunk 13 above)
Comment (vladfrangu) [reply stickyWallet/files/Configuration.kt:1]: Those are there for further reference once I can get off my lazy butt and migrate translations

File: src/main/kotlin/stickyWallet/StickyWallet.kt (hunk 1 above)
Comment (vladfrangu) [reply src/main/kotlin/stickyWallet/StickyWallet.kt:66]: Technically, yes, practically, I wouldn't need the command block in the build.gradle file. Then again, it's fine! :3

File: src/main/kotlin/stickyWallet/StickyWallet.kt (hunk 5 above)
Comment (vladfrangu) [reply src/main/kotlin/stickyWallet/StickyWallet.kt:1]: I mean you're right, it should've been `com.dumbdogdiner.stickyWallet`, however... the amount of nesting makes me wanna rage soo... If you want it, feel free to PR <3

File: src/main/kotlin/stickyWallet/sql/tables/AccountsTable.kt (hunk 10 above)
Comment (vladfrangu) [reply src/main/kotlin/stickyWallet/sql/tables/AccountsTable.kt:10]: _Yes_, but it doesn't affect it, it just says the max size CAN be 255. Forward compatibility 😄

**Pull request merged**